### ISSUE TITLE
Invoke-Sqlcmd does not parse correctly some string literal containing strings that resemble variables.

### ISSUE THREAD

**Issue by Matteo-T** [microsoft/SQLServerPSModule#22]
Run the following cmdlet:

```pwsh
invoke-sqlcmd -ServerInstance SQLTOOLS2022-3  -DisableVariables -Query "SELECT N'  ''`$(ESCAPE_SQUOTE(SRVR))'' '"
```

Result:
```console
Invoke-Sqlcmd:
Line |
   1 |  invoke-sqlcmd -ServerInstance SQLTOOLS2022-3  -DisableVariables -Quer …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Incorrect syntax was encountered while parsing '$(ESCAPE_SQUOTE('.
```

Expected:

```
Column1
-------
  '$(ESCAPE_SQUOTE(SRVR))'
```

**Comment by Matteo-T** [comment]
I need to pick up an updated ManagedParser with the fix for https://github.com/microsoft/sqltoolsservice/issues/1936 and wire up the new property.

**Comment by Matteo-T** [comment]
This is fixed in v22.0.52.2+ (when available)
